src/main/groovy/com/ullink/NuGetSpec.groovy

```diff
@@ -65,24 +61,54 @@ class NuGetSpec extends Exec {
     }
 
     String supplementDefaultValueOnNuspec(String nuspecString) {
+        def final msbuildTaskExists = project.tasks.findByName('msbuild') != null
+        def fwkVersions = [:]
+
+        if (msbuildTaskExists) {
+            project.tasks.withType(project.tasks.msbuild.getClass()).each {
+                def final csprojProperties = it.mainProject.properties
+                def final assemblyName = csprojProperties.AssemblyName.toString()
+                fwkVersions[assemblyName] = csprojProperties.TargetFrameworkVersion.toString().replace('v', '').replace('.', '')
+            }
+        }
 
         def root = new XmlSlurper(false, false).parseText(nuspecString)
 
         def defaultValues = {}
-        def applyDefaultValue = { String node, String value ->
+        def setDefaultMetadata = { String node, String value ->
             if (root.metadata[node].isEmpty()) {
                 defaultValues <<= { delegate."$node" value }
             }
         }
-        applyDefaultValue ('id', project.name)
-        applyDefaultValue ('version', project.version)
-        applyDefaultValue ('description', project.description)
+
+        setDefaultMetadata ('id', project.name)
+        setDefaultMetadata ('title', project.name)
+        setDefaultMetadata ('version', project.version)
+        setDefaultMetadata ('description', project.description ? project.description : project.name)
 
         if (root.metadata.isEmpty()) {
             root.appendNode { metadata defaultValues }
         } else {
             root.metadata.appendNode defaultValues
         }
+
+        if (msbuildTaskExists) {
+            def defaultFiles = {}
+            project.msbuild.mainProject.dotnetArtifacts.each {
+                artifact ->
+                    def fwkFolder = fwkVersions.find {
```

Comment:
I believe at the beginning I had in mind to handle multiple projects. It's not really needed and currently it doesn't work. If we want to do so I think we will need to do more work around here.
I've removed it.